From bd45ba6f48e9eedf51d4332afc24468b52fd5122 Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert@linux-m68k.org>
Date: Thu, 25 Jun 2015 15:10:32 +0200
Subject: [PATCH] flow_dissector: Pre-initialize ip_proto in
 __skb_flow_dissect()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit 8e690ffdbcc7b548aae3c644c3efcc6e0394fe37 upstream

net/core/flow_dissector.c: In function ‘__skb_flow_dissect’:
net/core/flow_dissector.c:132: warning: ‘ip_proto’ may be used uninitialized in this function

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
Acked-by: Tom Herbert <tom@herbertland.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[op: adjusted context for kernel version 3.4]
Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 net/core/flow_dissector.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/net/core/flow_dissector.c b/net/core/flow_dissector.c
index 9dd024c..108b918 100644
--- a/net/core/flow_dissector.c
+++ b/net/core/flow_dissector.c
@@ -23,7 +23,7 @@ static void iph_to_flow_copy_addrs(struct flow_keys *flow, const struct iphdr *i
 bool skb_flow_dissect(const struct sk_buff *skb, struct flow_keys *flow)
 {
 	int poff, nhoff = skb_network_offset(skb);
-	u8 ip_proto;
+	u8 ip_proto = 0;
 	bool ret = false;
 	__be16 proto = skb->protocol;
 
@@ -139,7 +139,6 @@ ipv6:
 			flow->ports = *ports;
 	}
 
-out_good:
 	ret = true;
 
 out_bad:
-- 
1.7.5.4

